src/line_shapes/get_line_shapes.jl

```diff
@@ -0,0 +1,128 @@
+
+"""
+Code to measure line bisectors and widths as a function of depth.
+Author: Alex Wise
+Created: September 2020
+"""
+
+""" get_line_shapes(line_list, clt, frac_depths, v_width, RV_estimate, verbose)
+For each spectrum in clt, returns arrays for
+line bisector and width as a function of fractional line depth given in frac_depths.
+Note that frac_depths must be sorted from largest to smallest.
+Also returns arrays for line depth, presently a simple minimum over the line's pixels, and valid lines,
+a boolean array for which lines' shape measurements were calculated successfully.
+v_width is the width of the window (in m/s) around each line's center to use.
+RV_estimate (in m/s) is a rough estimate of the star's RV.
+Note: RV_estimate shifts the line centers in line_list to the RV of star, so use RV_estimate=0 if line_list is already aligned to star.
+"""
+function get_line_shapes(line_list::A1, clt::AbstractChunkListTimeseries;
+                    frac_depths::A2 = 0.99:-0.01:0.05, v_width::Real=15e3,
+                    RV_estimate::Real=-5000.0, verbose::Bool = true) where { T1<:Real, A1<:AbstractArray{T1,1},
+                    T2<:Real, A2<:AbstractArray{T2,1}}
+
+    @assert issorted(frac_depths, rev=true) #make sure frac_depths are sorted largest to smallest
+    n_lines = length(line_list)
+    n_spectra = length(clt.chunk_list)
+    n_depths = length(frac_depths)
+    #allocate memory for line shapes
+    bisectors = zeros(n_depths,n_spectra,n_lines)
+    widths = zeros(n_depths,n_spectra,n_lines)
+    depths = zeros(n_spectra,n_lines)
+    valid_lines = ones(Bool,n_lines)
+    #find center of each chunk so that we can assign each line in line list to a chunk
+    chunk_centers = zeros(length(clt.chunk_list[1])) #note: this assumes all spectra have the same number of chunks
+    calc_chunk_centers(chunk_centers, clt.chunk_list[1]) # TODO: change chunk centers to average over all chunks in timeseries
+    #measure the bisectors and widths
+    for k in 1:n_spectra
+        line_shapes = get_line_shapes(line_list, clt.chunk_list[k], chunk_centers=chunk_centers,
+        frac_depths=frac_depths, v_width=v_width, RV_estimate=RV_estimate, verbose=verbose)
+        bisectors[:,k,:] = line_shapes[:bisectors]
+        widths[:,k,:] = line_shapes[:widths]
+        depths[k,:] = line_shapes[:depths]
+        valid_lines = valid_lines .& line_shapes[:valid_lines]
```

Comment:
Use `.=` to avoid allocating new array
`valid_lines .= valid_lines .& line_shapes[:valid_lines]`